lib/public/views/DataPasses/PerLhcPeriodOverview/DataPassesPerLhcPeriodOverviewModel.js

```diff
@@ -0,0 +1,122 @@
+import { SortModel } from '../../../components/common/table/SortModel.js';
+import { TextTokensFilterModel } from '../../../components/Filters/common/filters/TextTokensFilterModel.js';
+import { OverviewPageModel } from '../../../models/OverviewModel.js';
+
+/**
+ * Data Passes Per LHC Period overview model
+ */
+export class DataPassesPerLhcPeriodOverviewModel extends OverviewPageModel {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._overviewSortModel = new SortModel();
+        this._overviewSortModel.observe(() => {
+            this._pagination.silentlySetCurrentPage(1);
+            this.load();
+        });
+        this._overviewSortModel.visualChange$.bubbleTo(this);
+
+        this._namesFilterModel = new TextTokensFilterModel();
+        this._registerFilter(this._namesFilterModel);
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    async load({ lhcPeriodId } = {}) {
+        this._lhcPeriodId = lhcPeriodId ?? this._lhcPeriodId;
+        super.load();
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    getRootEndpoint() {
+        const params = {
+            filter: {
+                lhcPeriodIds: [this._lhcPeriodId],
+                names: this._namesFilterModel.normalized,
+            },
+        };
+
+        const { appliedOn: sortOn, appliedDirection: sortDirection } = this._overviewSortModel;
+        if (sortOn && sortDirection) {
+            params[`sort[${sortOn}]`] = sortDirection;
+        }
+
+        return buildUrl('/api/dataPasses', params);
+    }
+
+    /**
+     * Reset this model to its default
+     *
+     * @returns {void}
+     */
+    reset() {
+        this._namesFilterModel.reset();
+        super.reset();
+    }
+
+    /**
+     * Get id of current lhc period which data passes are fetched
+     */
+    get lhcPeriodId() {
+        return this._lhcPeriodId;
+    }
+
+    /**
+     * Returns the model handling the overview page table sort
+     *
+     * @return {SortModel} the sort model
+     */
+    get overviewSortModel() {
+        return this._overviewSortModel;
+    }
+
+    /**
+     * Returns data passes names filter model
+     * @return {TextTokensFilterModel} data passes names filter model
+     */
+    get namesFilterModel() {
+        return this._namesFilterModel;
+    }
+
+    /**
+     * Register a new filter model
+     * @param {FilterModel} filterModel the filter model to register
+     * @return {void}
+     * @private
+     */
+    _registerFilter(filterModel) {
+        filterModel.visualChange$.bubbleTo(this);
+        filterModel.observe(() => {
+            this._pagination.silentlySetCurrentPage(1);
+            this.load();
+        });
+    }
+
+    /**
+     * States whether any filter is active
+     * @return {boolean} true if any filter is active
+     */
+    isAnyFilterActive() {
+        return !this._namesFilterModel.isEmpty();
```

Comment:
Nothing about perios id?

Reply:
Yes, because periodId is not subject for filtering, it plays the same role as lhcPeriodName in RunsPerLhcPeriod view